tests/basic/storage_pool_test.go

```diff
@@ -8483,3 +8500,120 @@ var _ = Describe("{DriveAddAsJournal}", func() {
 	})
 
 })
+
+var _ = Describe("{ReplResyncOnPoolExpand}", func() {
+	/*
+		PTX-15696 -> PWX-26967
+		Deploy IO aggressive application using repl-2 volumes
+		Identify the pools of this volume
+		Invoke pool expand is one pool of this volume and wait for pool expand to be completed
+		Volume status will be degraded when pool expand is going on for one of the pool
+		Volume repl resync should not fail after pool expand is done (This behavior after fix)
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("ReplResyncOnPoolExpand",
+			"Resync failed for a volume after pool came out of rebalance",
+			nil, 0)
+	})
+
+	Inst().AppList = []string{}
+	var ioIntensiveApp = []string{"fio", "fio-writes"}
+
+	for _, eachApp := range ioIntensiveApp {
+		Inst().AppList = append(Inst().AppList, eachApp)
+	}
+	var contexts []*scheduler.Context
+	stepLog := "Resync volume after rebalance"
+	It(stepLog, func() {
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("replresyncpoolexpand-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		// Get a pool with running IO
+		poolUUID, err := GetPoolIDWithIOs(contexts)
+		log.FailOnError(err, "Failed to get pool running with IO")
+		log.InfoD("Pool UUID on which IO is running [%s]", poolUUID)
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+
+		// Get All Volumes from the pool
+		volumes, err := GetVolumesFromPoolID(contexts, poolUUID)
+		log.FailOnError(err, "Failed to get list of volumes from the poolIDs")
+
+		// Change replication factor to 2 on all the volumes
+		volumeReplicaMap := make(map[string]int)
+		revertReplica := func() {
+			for _, eachvol := range volumes {
+				for volName, replcount := range volumeReplicaMap {
+					if eachvol.Name == volName {
+						getReplicaSets, err := Inst().V.GetReplicaSets(eachvol)
+						log.FailOnError(err, "Failed to get replication factor on the volume")
+						if len(getReplicaSets[0].Nodes) != replcount {
+							err := Inst().V.SetReplicationFactor(eachvol, 2, nil, nil, true)
+							log.FailOnError(err, "failed to set replicaiton value of Volume [%v]", volName)
+						}
+					}
+				}
+			}
+		}
+
+		defer revertReplica()
+		for _, eachVol := range volumes {
+			getReplicaSets, err := Inst().V.GetReplicaSets(eachVol)
+			log.FailOnError(err, "Failed to get replication factor on the volume")
+			volumeReplicaMap[eachVol.Name] = len(getReplicaSets[0].Nodes)
+
+			if len(getReplicaSets[0].Nodes) != 2 {
+				err := Inst().V.SetReplicationFactor(eachVol, 2, nil, nil, true)
+				if err != nil {
```

Comment:
if is not required here

Reply:
oops I missed it updating ! will fix it along with other review that I send for review